compiler/resolution/preFold.cpp

```diff
@@ -1884,6 +1885,81 @@ static Expr* preFoldPrimOp(CallExpr* call) {
     break;
   }
 
+  case PRIM_RESOLVES: {
+    if (call->id == breakOnResolveID) gdbShouldBreakHere();
+
+    // This primitive should only have one actual.
+    INT_ASSERT(call && call->numActuals() == 1);
+
+    auto exprToResolve = call->get(1);
+
+    // TODO: To resolve an arbitrary expression, we need a way to back out
+    // of resolving any expression without emitting errors. This is
+    // probably easier to do in a new compiler world.
+    if (!isCallExpr(exprToResolve)) {
+      INT_FATAL("PRIM_RESOLVES can only resolve CallExpr for now");
+    }
```

Comment:
Doesn't the code below already handle this? At least it handles nested calls, right?

Reply:
I added a `(e.g. a SymExpr)` and noted that we'd have to interact with `scopeResolve` as well.

Reply:
Do you mean `UnresolvedSymExpr` ? Because a SymExpr is already resolved, by nature.